web/src/components/game/goals/GoalManagement.tsx

```diff
@@ -149,6 +196,38 @@ export default function GoalManagement() {
                     />
                 </DialogContent>
             </Dialog>
+
+            {/* Delete Confirmation Dialog */}
+            <Dialog ref={deleteDialogRef}>
```

Comment:
If a dialog requires it's own standalone state, it should usually be it's own component